Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2512,17 +2512,22 @@ RefPtr<ByteArrayPixelBuffer> CanvasRenderingContext2DBase::cacheImageDataIfPossi
     return cachedBuffer;
 }
 
-RefPtr<ImageData> CanvasRenderingContext2DBase::makeImageDataIfContentsCached(const IntRect& sourceRect, PredefinedColorSpace colorSpace) const
+RefPtr<ImageData> CanvasRenderingContext2DBase::makeImageDataIfContentsCached(const IntRect& sourceRect, PredefinedColorSpace colorSpace, ImageDataStorageFormat storageFormat) const
```

Comment:
I think it would be fine to make this caching mechanism work only for Uint8 storage, and then not worry about passing the storage format value in.